### PR TITLE
Networkd: Add route to configurable route table

### PR DESCRIPTION
**Pull request by jkoelker**
Updated the DHCP and IPv6AcceptRA sections to allow setting the table the routes will be added to with the RouteTable key.

### CODE REVIEW

**Comment by bhouse** [comment]
@jkoelker I was just searching for this functionality and found https://github.com/systemd/systemd/pull/3132, not sure if that meets your needs too but it looks similar to this change.

**Comment by jkoelker** [comment]
@bhouse #3132 allows setting the table for static routes. This allows setting the route table for routes obtained via DHCP and IPV6 Router Advertisements and uses the underlying functionality from that patch.

**Comment by tr3buchet** [comment]
+1

File: man/systemd.network.xml [hunk 1]
```diff
@@ -917,6 +917,15 @@
             DHCP server.</para>
           </listitem>
         </varlistentry>
+
+        <varlistentry>
+          <term><varname>RouteTable=<replaceable>num</replaceable></varname></term>
+          <listitem>
+            <para>The table identifier for dhcp routes (a number between 1 and 4294967295, or 0 to unset).
```
Comment (keszybz) [inline man/systemd.network.xml:924]: dhcp → DHCP

File: man/systemd.network.xml [hunk 2]
```diff
@@ -966,6 +966,15 @@
             project='man-pages'><refentrytitle>resolv.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
           </listitem>
         </varlistentry>
+
+        <varlistentry>
+          <term><varname>RouteTable=<replaceable>num</replaceable></varname></term>
+          <listitem>
+            <para>The table identifier for dhcp routes (a number between 1 and 4294967295, or 0 to unset).
```
Comment (keszybz) [inline man/systemd.network.xml:973]: DHCP

**Pull request merged**